src/webapp/components/table/performance-overview-table/PerformanceOverviewTable.tsx

```diff
@@ -0,0 +1,277 @@
+import React, { useEffect, useMemo, useState } from "react";
+import _ from "lodash";
+import {
+    Table,
+    TableBody,
+    TableCell,
+    TableHead,
+    TableRow,
+    TableContainer,
+} from "@material-ui/core";
+import styled from "styled-components";
+import { SearchInput } from "../../search-input/SearchInput";
+import { Selector } from "../../selector/Selector";
+import { Maybe } from "../../../../utils/ts-utils";
+import i18n from "../../../../utils/i18n";
+
+export type TableColumn = {
+    value: string;
+    label: string;
+    dark?: boolean;
+};
+
+interface PerformanceOverviewTableProps {
+    rows: {
+        [key: TableColumn["value"]]: string;
+    }[];
+}
+
+export const PerformanceOverviewTable: React.FC<PerformanceOverviewTableProps> = React.memo(
+    ({ rows }) => {
+        const [searchTerm, setSearchTerm] = useState<string>("");
+        const [filterValue, setFilterValue] = useState("");
+        const [filteredRows, setFilteredRows] = useState(rows);
+
+        const columns: TableColumn[] = [
+            { label: "Event", value: "event" },
+            { label: "Location", value: "location" },
+            { label: "Cases", value: "cases" },
+            { label: "Deaths", value: "deaths" },
+            { label: "Duration", value: "duration" },
+            { label: "Manager", value: "manager" },
+            { label: "Detect 7d", dark: true, value: "detect7d" },
+            { label: "Notify 1d", dark: true, value: "notify1d" },
+            { label: "ERA1", value: "era1" },
+            { label: "ERA2", value: "era2" },
+            { label: "ERA3", value: "era3" },
+            { label: "ERA4", value: "era4" },
+            { label: "ERA5", value: "era5" },
+            { label: "ERA6", value: "era6" },
+            { label: "ERA7", value: "era7" },
+            { label: "ERI", value: "eri" },
+            { label: "Respond 7d", dark: true, value: "respond7d" },
+        ];
+
+        const editRiskAssessmentColumns = [
+            "era1",
+            "era2",
+            "era3",
+            "era4",
+            "era5",
+            "era6",
+            "era7",
+            "eri",
+        ];
+        const columnRules: { [key: string]: number } = {
+            detect7d: 7,
+            notify1d: 1,
+            respond7d: 7,
+        };
+
+        const calculateColumns = [...editRiskAssessmentColumns, ..._.keys(columnRules)];
+
+        useEffect(() => {
```

Comment:
maybe move this to a custom hook like useTableSearch?

Reply:
Should I move custom hook into a specific file?

Reply:
When the custom hook is generic I usually add it to a file in` webapp > hooks` or also I have seen in some project that they are added to `webapp > common > hooks`. But if it's a custom hook of the component I would added to the same folder with the component. In this PR https://github.com/EyeSeeTea/zebra-dev/pull/5 I have added both type of hooks (but it's still in draft :pray: )